libs/s25main/network/GameClient.cpp

```diff
@@ -323,8 +323,10 @@ void GameClient::StartGame(const unsigned random_init)
             gameWorld.GetPlayer(i).MakeStartPacts();
 
         MapLoader loader(gameWorld);
-        if(!loader.Load(mapinfo.filepath)
-           || (!mapinfo.luaFilepath.empty() && !loader.LoadLuaScript(*game, *this, mapinfo.luaFilepath)))
+        if((!mapinfo.luaFilepath.empty() && !loader.LoadLuaScript(*game, *this, mapinfo.luaFilepath))
+           || !loader.Load(mapinfo.filepath)) // Do not reorder: load lua first, load map second.
+                                              // If the map is loaded first and it does not have a player HQ set, it
+                                              // will not load correctly, even though the HQ may be set using LUA.
```

Comment:
I still find it hard to understand why it will "not load correctly" and what loading the script first changes. What about:

```suggestion
        // Load (and verify) Lua script before the map.
        // If a valid script exists loading the map will not abort when HQ positions in the map are missing
        // as HQs might be placed by the script when starting the game.
        if((!mapinfo.luaFilepath.empty() && !loader.LoadLuaScript(*game, *this, mapinfo.luaFilepath))
           || !loader.Load(mapinfo.filepath))
```